addons/misc/functions/fnc_suitcaseNukeZeus.sqf

```diff
@@ -0,0 +1,211 @@
+                    _explosive setDamage 1;
+                    deleteVehicle _device;
+                };
+                case "explosive_large": {
+                    private _explosive = createVehicle ["Bo_GBU12_LGB", getPosATL _device, [], 0, "CAN_COLLIDE"];
+                    _explosive setDamage 1;
+                    deleteVehicle _device;
+                };
+                case "explosive_nuke": {
+                    ["zen_modules_moduleNuke", [ASLToAGL(getPosASL _unit), 200, 300, false]] call CBA_fnc_globalEvent;
+                    deleteVehicle _device;
+                };
+                case "smoke_yellow": {
+                    private _smoke = createVehicle ["SmokeShellYellow_Infinite", getPosATL _device, [], 0, "CAN_COLLIDE"];
+                    hideObjectGlobal _smoke;
+                    _device setVariable [QGVAR(suitcaseNuke_smoke), _smoke];
+                    _smoke attachTo [_device];
+                    _device addEventHandler ["Deleted", {
+                        params ["_device"];
+                        deleteVehicle (_device getVariable [QGVAR(suitcaseNuke_smoke), objNull]);
+                    }];
+                };
+                default {};
+            };
+        };
+
+        // private _customCode = _device getVariable [QGVAR(suitcaseNuke_code), ""];
+        // if(_customCode isNotEqualTo "") then {
+        //     private _codeTarget = _unit getVariable [QGVAR(suitcaseNuke_codeTarget), 0];
+
+        //     if((_codeTarget == 0 and hasInterface) || {
+        //         (_codeTarget == 1 and isServer) || {
+        //         (_codeTarget == 2)}}
+        //     ) then {
+        //         [_device] call (compile _customCode);
+        //     };
+        // };
+    }];
+
+
+    // Defusal Code
+    if(_defuser > 0) then {
+
+        // TODO: with ace, also (optionally) require DefusalKit item
+        private _ace = (isClass(configFile >> "CfgPatches" >> "ace_main"));
+        private _condition = switch(true) do {
+            case (_defuser == 1 and _ace) : { " and ([player] call ace_common_fnc_isEOD)" }; // EOD
+            case (_defuser == 1 and !_ace) : { " and (player getUnitTrait ""ExplosiveSpecialist"")" }; // EOD
+            case (_defuser == 2 and _ace) : { " and (([player] call ace_common_fnc_isEngineer) or ([player] call ace_common_fnc_isEOD)" }; // Engineer
+            case (_defuser == 2 and !_ace) : { " and ((player getUnitTrait ""Engineer"") or (player getUnitTrait ""ExplosiveSpecialist""))" }; // Engineer
+            default { "" }; // Anyone
+        };
+
+
+        [
+            _device,
+            "Defuse Device",
+            "\a3\ui_f_oldman\data\IGUI\Cfg\holdactions\repair_ca.paa",
+            "\a3\ui_f_oldman\data\IGUI\Cfg\holdactions\repair_ca.paa",
+            format ["(_this distance _target < 3) and (alive _target) and _target getVariable [""crowsza_misc_suitcaseNuke_isArmed"", false]%1", _condition],
```

Comment:
please use ``QGVAR(suitcaseNuke_isArmed)`` as that will automatically add the prefixes and put it in quotes. I think you can even do:
```
QUOTE((_this distance _target < 3) and (alive _target) and _target getVariable [QGVAR(crowsza_misc_suitcaseNuke_isArmed), false]%1)
```

Reply:
I ran into an issue doing that (I think it may have been in an earlier PR) where using setVariable inside a QUOTE caused some odd behaviour when it hit the comma; I'll double check on that and see if I can clean this up!

Reply:
Okay, so:

`QUOTE((_this distance _target < 3) and (alive _target) and _target getVariable [QQGVAR(suitcaseNuke_isArmed), false])` fails to be processed - the `QUOTE(...)` gets replaced with an empty string.

`QUOTE((_this distance _target < 3) and (alive _target) and (_target getVariable [QQGVAR(suitcaseNuke_isArmed), false]))` fails as well - the `,` before false gets removed from the processed string

`QUOTE((_this distance _target < 3) and (alive _target) and _target getVariable [QQGVAR(suitcaseNuke_isArmed)])` gets processed just fine, but I'd rather be explicit about what the default behaviour is

Reply:
...but doing `#define COMMA ,` and then

`QUOTE((_this distance _target < 3) and (alive _target) and (_target getVariable [QQGVAR(suitcaseNuke_isArmed) COMMA false]))` sorts it

Reply:
Might be needed to use ARR macro for the array? 
![billede](https://github.com/Crowdedlight/Crows-Zeus-Additions/assets/7889925/f1c2ce90-321b-496e-b901-d3658609afc1)

https://discord.com/channels/976165959041679380/1039189045655392389/1204464769373835295 (Ace Discord)

But I also saw this lately. Is that the same thing you are running into? 
![billede](https://github.com/Crowdedlight/Crows-Zeus-Additions/assets/7889925/17f39b82-112a-4f33-a28d-1e2389ea0e04)

Reply:
Oh, yeah, looks like exactly that issue - spooky. ARR might fix it, will try